src/components/AppComponent.js

```diff
@@ -66,7 +66,7 @@ export const AppComponent = connect(({ dataNonce, focus, search, user, settings,
   })}>
     <Sidebar />
     <HamburgerMenu />
-    <MultiGesture onGesture={handleGestureSegment} onEnd={handleGestureEnd}>
+    <MultiGesture showOverlay={showOverlay} onGesture={handleGestureSegment} onEnd={handleGestureEnd}>
```

Comment:
Correct me if I am wrong, but I believe if you pass `showOverlay` as a prop, then when it changes it will re-render the entire component. Instead, I would suggest accessing the store directly in the gesture and keyboard handlers.

Reply:
Ok